cvat/apps/events/views.py

```diff
@@ -0,0 +1,151 @@
+
+
+from cvat.apps.events.serializers import ClientEventsSerializer, EventSerializer
+from cvat.apps.engine.log import clogger, vlogger
+from .export import export
+
+class EventsViewSet(viewsets.ViewSet):
+    serializer_class = None
+    TIME_THRESHOLD = 100 # seconds
+
+    @staticmethod
+    @extend_schema(summary='Method saves logs from a client on the server',
+        methods=['POST'],
+        description='Sends logs to the Clickhouse if it is connected',
+        request=ClientEventsSerializer(),
+        responses={
+            '201': ClientEventsSerializer(),
+        })
+    @extend_schema(summary='Method returns csv log file ',
+        methods=['GET'],
+        description='Recieve logs from the server',
+        parameters=[
+            OpenApiParameter('org', location=OpenApiParameter.QUERY, type=OpenApiTypes.INT, required=False,
+                description="Filter events by organization ID"),
+            OpenApiParameter('project', location=OpenApiParameter.QUERY, type=OpenApiTypes.INT, required=False,
+                description="Filter events by project ID"),
+            OpenApiParameter('task', location=OpenApiParameter.QUERY, type=OpenApiTypes.INT, required=False,
+                description="Filter events by task ID"),
+            OpenApiParameter('job', location=OpenApiParameter.QUERY, type=OpenApiTypes.INT, required=False,
+                description="Filter events by job ID"),
+            OpenApiParameter('user', location=OpenApiParameter.QUERY, type=OpenApiTypes.INT, required=False,
+                description="Filter events by user ID"),
+            OpenApiParameter('from', location=OpenApiParameter.QUERY, type=OpenApiTypes.DATETIME, required=False,
+                description="Filter events after the datetime"),
+            OpenApiParameter('to', location=OpenApiParameter.QUERY, type=OpenApiTypes.DATETIME, required=False,
+                description="Filter events before the datetime"),
+            OpenApiParameter('filename', description='Desired output file name',
+                location=OpenApiParameter.QUERY, type=OpenApiTypes.STR, required=False),
+            OpenApiParameter('action', location=OpenApiParameter.QUERY,
+                description='Used to start downloading process after annotation file had been created',
+                type=OpenApiTypes.STR, required=False, enum=['download']),
+            OpenApiParameter('query-id', location=OpenApiParameter.QUERY, type=OpenApiTypes.STR, required=False,
+                description="ID of query request that need to check or download"),
+        ],
+        responses={
+            '200': OpenApiResponse(description='Download of file started'),
+            '201': OpenApiResponse(description='CSV log file is ready for downloading'),
+            '202': OpenApiResponse(description='Creating a CSV log file has been started'),
+        })
+    @action(detail=False, methods=['GET', 'POST'])
+    def events(request):
```

Comment:
Why do we need the action in EventsViewSet? Why don't use 'create' and 'list' methods?

Reply:
Also, we need to think about the interface for long operations. For example, how to get the list of all operations which are in progress now? Let's say we reload UI, we don't know query_id anymore. How to download the CSV file?